### PR TITLE
no need to override end

### PR DESCRIPTION
**Pull request by calvinmetcalf**
Node 8.1.0 includes a new method just for cases like this where you need to clean up at the end of a writable stream, this pull switches over to use the new `_final` method and switches to use readable-stream instead of stream so it'll work across all versions of node

### CODE REVIEW

File: test/unit/upload-stream-unit.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ var expect = chai.expect;
 var UploadStream = require("../../lib/upload-stream");
 var Uploader = require("../../lib/uploader");
 var S3 = require("mock-s3").S3;
-var Writable = require("stream").Writable;
+var Writable = require("readable-stream").Writable;
```
Comment (calvinmetcalf) [inline test/unit/upload-stream-unit.js]: we can also just get rid of the test that checks inheritance of the stream object if you want

**Comment by jsantell** [comment]
This looks good, thanks for this! Could you also add Node 4, 6 and 8 into the .travis.yml just for sanity checks it works on there? It's been awhile since that has been updated :)

**Comment by calvinmetcalf** [comment]
done

**Comment by jsantell** [comment]
@calvinmetcalf pushed as [2.0.2](https://www.npmjs.com/package/s3-stream-upload)

**Pull request merged**